### PR TITLE
new: implement an async thread for generating diffs

### PR DESCRIPTION
**Pull request by Andreagit97**

### CODE REVIEW

File: src/async.cpp [hunk 1]
```diff
@@ -179,7 +181,20 @@ void my_plugin::async_thread_loop(std::unique_ptr<falcosecurity::async_event_han
...
+
+void my_plugin::diff_notify(std::unique_ptr<falcosecurity::async_event_handler> h) noexcept {
+	SPDLOG_INFO("Start diff notify thread");
+	while(!m_diff_notify_thread_quit.load()) {
+		// -1 should be a valid tid for async events
+		generate_async_event(h, ASYNC_EVENT_DIFF_NAME, -1, "root");
+		std::unique_lock<std::mutex> l(m_diff_mu);
+		m_diff_cv.wait_for(l, std::chrono::seconds(DIFF_NOTIFY_INTERVALS), [this] {
+			return m_diff_notify_thread_quit.load();
+		});
```
Comment (irozzo-1A) [inline src/async.cpp:195]: nit: you could avoid using an atomic by doing the following
```suggestion
        bool done = false;
	while(done) {
		// -1 should be a valid tid for async events
		generate_async_event(h, ASYNC_EVENT_DIFF_NAME, -1, "root");
		std::unique_lock<std::mutex> l(m_diff_mu);
		done = m_diff_cv.wait_for(l, std::chrono::seconds(DIFF_NOTIFY_INTERVALS), [this] {
			return m_diff_notify_thread_quit;
		});
```

**Review by irozzo-1A — APPROVED** [review]
LGTM just a nit

**Comment by FedeDP** [comment]
My idea was to use a timerfd in the same poll async thread we already have.
Also in m_cfg we already have a userspecified timeout in seconds exactly for this event.

**Comment by FedeDP** [comment]
My idea was to use a timerfd in the same poll async thread we already have.
Also in m_cfg we already have a underspecified timeout in seconds exactly for this event.

I've: I'd avoid using another thread while we can just reuse the same one we already have.

**Review by FedeDP — CHANGES_REQUESTED** [review]
My take: https://github.com/FedeDP/troublescope/pull/16

**Pull request closed without merging**